### PR TITLE
fixed_mult change

### PR DESCRIPTION
**Pull request by ponut64**
(really just jo_fixed_mult change. if you want more streamlined fixed-point math in assembly, give what you need, i can produce)

### CODE REVIEW

File: jo_engine/math.c [hunk 1]
```diff
@@ -1,77 +1,68 @@
...
+#include "jo/math.h"
 
 /*
 ** FIXED Q16.16 Number
 */
 
+//Dear Johannes,
+//I cannot let this crime against performant code stand.
+//Allow me this *one* transgression?
+// - Ponut64
```
Comment (johannes-fetz) [inline jo_engine/math.c:50]: Sure :)

**Comment by johannes-fetz** [comment]
Can you just commit your modification and I validate the merge ?

**Comment by johannes-fetz** [comment]
@ponut64 I did not see that the code was not compiling

**Comment by johannes-fetz** [comment]
..\..\jo_engine\math.c|65|error: undefined named operand 'd1'|
..\..\jo_engine\math.c|65|error: undefined named operand 'd2'|

**Comment by johannes-fetz** [comment]
@ponut64 I revert the merge to let you fix the compilation

**Comment by ponut64** [comment]
maybe i should actually check things before i make stupi

**Pull request merged**